### PR TITLE
move kyc module to shared modules

### PR DESCRIPTION
**Pull request by huesforalice**
This PR does the following:

- Moves kyc module (incl IdentityRegistry contract) to shared
- Creates a new instant id module in the web project that takes care of onfido and id-now
- Unfortunately moves some translations into the shared modules (see review notes)

### CODE REVIEW

File: packages/shared-modules/src/lib/yup/custom-schemas.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import { ECountries } from "@neufund/shared-utils";
+import { includes } from "lodash";
+import moment from "moment";
+import * as Yup from "yup";
+
+/**
+ * Custom schema translations are baked in to this file, as this code is
+ * outdated they will go soon.
+ */
+const VALIDATION_CURRENCY_CODE = "Must be a 3-letter currency code";
```
Comment (huesforalice) [inline packages/shared-modules/src/lib/yup/custom-schemas.ts:10]: Unfortunately I had to move the translations here, it's not a very good solution so I'm happy for suggestions

File: packages/shared-modules/src/modules/routing/actions.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { createActionFactory } from "@neufund/shared-utils";
```
Comment (huesforalice) [inline packages/shared-modules/src/modules/routing/actions.ts:1]: This routes will have to be forwarded in the respective projects. It's possible that we will add some more generic names for them, but for now it should do.

File: packages/wallet/app/modules/contracts/lib/IdentityRegistryAdapter.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * An adapter for RateOracle to hide implementation differences between web3 and ethers generated contracts interfaces
+ */
+class IdentityRegistryAdapter implements IIdentityRegistryAdapter {
+  readonly address: string;
+
+  constructor(private readonly identityRegistry: IIdentityRegistry) {
+    this.address = this.identityRegistry.address;
+  }
+  async getClaims(address: string): Promise<string> {
```
Comment (huesforalice) [inline packages/wallet/app/modules/contracts/lib/IdentityRegistryAdapter.ts:25]: Can this be done in a better way?

File: packages/web/app/modules/instant-id/id-now/sagas.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  TKycIdNowIdentification,
+} from "@neufund/shared-modules";
+
+import { createMessage } from "../../../components/translatedMessages/utils";
+import { TGlobalDependencies } from "../../../di/setupBindings";
+import { actions } from "../../actions";
+import { neuGetContext, neuTakeEvery } from "../../sagasUtils";
+
+function* startIdNow(_: TGlobalDependencies): Generator<any, any, any> {
+  const { apiKycService, logger } = yield* neuGetContext();
```
Comment (huesforalice) [inline packages/web/app/modules/instant-id/id-now/sagas.ts:15]: Dep resolution needs to be done this way, as at the time of module instantiation the deps context object is not available.

File: packages/shared-modules/src/modules/kyc/module.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  const module = {
+    id: MODULE_ID,
+    api: kycApi,
+    libs: [setupContainerModule()],
+    sagas: [setupKycSagas(config)],
+    reducerMap: kycReducerMap,
+  };
+
+  return [
+    setupAuthModule({
```
Comment (huesforalice) [inline packages/shared-modules/src/modules/kyc/module.ts:30]: This is not very nice, but I was not sure how to do it ny other way.

File: packages/web/app/modules/routing/sagas.ts [hunk 6]
```diff
@@ -110,7 +110,71 @@ export function* routeAction<P>(
   }
 }
 
+/**
+ * Forward routes from shared modules
+ */
+export function* sharedRoutes(): Generator<unknown, void> {
+  // default routes
+  yield fork(neuTakeEvery, routingModuleApi.actions.goToDashboard, function*(): Generator<
```
Comment (desfero) [inline packages/web/app/modules/routing/sagas.ts:118]: For new sagas we should use a proper type `SagaGenerator<void>`

File: packages/web/app/store.ts [hunk 7]
```diff
@@ -79,8 +83,14 @@ export const setupAppModule = ({ history, config, container }: TAppModuleConfig)
     ...setupWebTxHistoryModule({
       refreshOnAction: actions.web3.newBlockArrived,
     }),
+    setupInstantIdModule(),
```
Comment (desfero) [inline packages/web/app/store.ts:86]: given that instant id uses kyc under the hood it should declare it as a dependency.

**Comment by desfero** [comment]
Please make sure changes in https://github.com/Neufund/platform-frontend/pull/4425/files are applied properly.

File: packages/shared-modules/src/modules/kyc/module.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  ];
+};
+
+const kycApi = {
+  actions: kycActions,
+  selectors,
+  sagas: { loadKycRequestData, loadBankAccountDetails, loadKycStatus },
+  utils,
+  symbols,
+  reducerMap: kycReducerMap,
```
[the remark this answers is not in the document]
Comment (huesforalice) [reply packages/shared-modules/src/modules/kyc/module.ts:44]: It's used in the tests for the wallet view sagas on the web project..

File: packages/web/app/modules/auth/jwt/sagas.ts [hunk 9]
```diff
@@ -21,7 +21,7 @@ import { JwtNotAvailable, MessageSignCancelledError } from "../errors";
  * If needed permissions are not present/valid will escalate permissions with authentication server
  */
 export function* ensurePermissionsArePresentAndRunEffect(
-  { logger }: TGlobalDependencies,
+  deps: TGlobalDependencies,
```
[the remark this answers is not in the document]
Comment (huesforalice) [reply packages/web/app/modules/auth/jwt/sagas.ts:24]: I could not get the typings to work for the message signature in the shared code when the deconstruction argument was there unfortunately..

File: packages/web/app/modules/instant-id/id-now/selectors.ts [hunk 10]
```diff
@@ -1,7 +1,7 @@
 import { createSelector } from "reselect";
 
-import { TAppGlobalState } from "../../../../store";
+import { TAppGlobalState } from "../../../store";
 
-export const selectKycIdNow = (state: TAppGlobalState) => state.kyc.idNow;
+export const selectKycIdNow = (state: TAppGlobalState) => state.instant_id.idNow;
```
[the remark this answers is not in the document]
Comment (huesforalice) [reply packages/web/app/modules/instant-id/id-now/selectors.ts:5]: Haha yeah :) Will fix it!

File: packages/shared-modules/package.json [hunk 11]
```diff
@@ -65,6 +65,7 @@
     "@neufund/sagas": "^1.0.0",
     "@neufund/shared-utils": "^1.0.0",
     "bignumber.js": "^5.0.0",
+    "moment": "^2.21.0",
```
[the remark this answers is not in the document]
Comment (huesforalice) [reply packages/shared-modules/package.json:68]: done!

File: packages/shared-modules/src/modules/kyc/module.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (huesforalice) [reply packages/shared-modules/src/modules/kyc/module.ts:44]: Maybe we should have some kind of "tests export" for these cases?

File: packages/web/app/modules/instant-id/id-now/sagas.ts (hunk 4 above)
Comment (huesforalice) [reply packages/web/app/modules/instant-id/id-now/sagas.ts:15]: done

File: packages/wallet/app/modules/contracts/lib/IdentityRegistryAdapter.ts (hunk 3 above)
Comment (huesforalice) [reply packages/wallet/app/modules/contracts/lib/IdentityRegistryAdapter.ts:25]: Done!

File: packages/web/app/modules/routing/sagas.ts (hunk 6 above)
Comment (huesforalice) [reply packages/web/app/modules/routing/sagas.ts:118]: Done

File: packages/web/app/modules/instant-id/onfido/sagas.ts [hunk 12]
```diff
@@ -1,27 +1,25 @@
...
 type TChannelTypes = {
   type: EOnfidoSDKEvents;
 };
 
-function* initOnfidoSdk({
-  onfidoSDK,
-  apiKycService,
-}: TGlobalDependencies): Generator<any, any, any> {
+function* initOnfidoSdk(_: TGlobalDependencies): Generator<any, any, any> {
+  const { apiKycService, onfidoSDK } = yield* neuGetContext();
```
[the remark this answers is not in the document]
Comment (huesforalice) [reply packages/web/app/modules/instant-id/onfido/sagas.ts:14]: done

File: packages/web/app/store.ts (hunk 7 above)
Comment (huesforalice) [reply packages/web/app/store.ts:86]: Yeah, I would prefer to update this when we know how to resolve the module dependencies properly, I've left a comment for now.

File: packages/web/app/components/eto/registration/pages/EtoPitch/validateEtoRegistrationPitch.tsx [hunk 13]
```diff
@@ -1,3 +1,4 @@
+import { customSchemas } from "@neufund/shared-modules";
```
[the remark this answers is not in the document]
Comment (huesforalice) [reply packages/web/app/components/eto/registration/pages/EtoPitch/validateEtoRegistrationPitch.tsx:1]: so you want to move the yup stuff to shared utils, is that right?

[the remark this answers is not in the document]
Comment (desfero) [reply packages/web/app/components/eto/registration/pages/EtoPitch/validateEtoRegistrationPitch.tsx:1]: exactly, but can be done later

**Pull request merged**